src/alchemlyb/parsing/gmx.py

```diff
@@ -105,6 +105,9 @@ def extract_u_nk(xvg, T):
 
     u_k.name = 'u_nk'
 
+    u_k.attrs['temperature'] = T
+    u_k.attrs['energy_unit'] = 'kT'
+
```

Comment:
We could wrap attribute setting so that we can use old pandas. 
```python
# remove once we can require pandas >= 1.2 (after Python 3.
if pd.__version__ >= "1.2":   # do this version check properly
   def pd_setattrs(df, data):
         df.attrs = data
else:
   def pd_setattrs(df, data):
        if not isinstance(data, dict):
              raise TypeError
        with warnings.catch_warnings():
              warnings.simplefilter("ignore")
              setattr(df, 'attrs', data) 
```
and then use `pd_setattrs()` everywhere where you set the attributes for the first time.

Reply:
This could live in util or somewhere similar.

Reply:
@orbeckst Sorry, it is a bit unclear to me of what this function is trying to achieve. There should be no problem if one were to only use the alchemlyb functions when pandas < 1.2, but if one were to manipulate the data frame by themselves, pandas < 1.2 would then drop the attrs.

Reply:
When I tried to set `df.attrs`, pandas complained that I was trying to add a column but set the attribute nevertheless. My idea was to just filter the warnings.

It was not clear to me what the problem with attrs for 1.0 ≤ pandas < 1.2 was: Are you saying that only for ≥ 1.2, slicing a df will retain the attrs on the sliced copy?

Reply:
@orbeckst 

> When I tried to set df.attrs, pandas complained that I was trying to add a column but set the attribute nevertheless. My idea was to just filter the warnings.

For pandas == 1.0.5,
```
>>> d = {'col1': [1, 2], 'col2': [3, 4]}
>>> df = pd.DataFrame(data=d)
>>> df.attrs
{}
>>> df.attrs = {1:1}
```
yields no warning for me.

> It was not clear to me what the problem with attrs for 1.0 ≤ pandas < 1.2 was: Are you saying that only for ≥ 1.2, slicing a df will retain the attrs on the sliced copy?

Yes.